### PR TITLE
Migrate low code editor LS extensions to alpha5

### PR DESCRIPTION
**Pull request by kaviththiranga**
## Purpose
Includes several fixes to lang-server extensions for low code editor

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/module-info.java [hunk 1]
```diff
@@ -74,5 +74,6 @@
     exports io.ballerina.projects.plugins;
     exports io.ballerina.projects.internal.model; // TODO Remove this exports
     exports io.ballerina.projects.internal.environment; // TODO Remove these exports
+    exports io.ballerina.compiler.api.impl.symbols;
```
Comment (nadeeshaan) [inline compiler/ballerina-lang/src/main/java/module-info.java:77]: It is not the intended usage to utilize the implementation classes. Instead we use the exposed public interfaces. I see that the reason to use the impl is to use `BallerinaUnionTypeSymbol` in `BallerinaSymbolServiceImpl.java`. Here we can use the `UnionTypeSymbol` which is the public interface. Hence shall we remove this export?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java [hunk 2]
```diff
@@ -105,24 +111,29 @@ private boolean isWithinLineRange(LineRange containerRange, LineRange childRange
...
 
     private void moveUnusedtoUsedImport(LineRange lineRange, ImportDeclarationNode importDeclarationNode) {
-        List<Location> locations = this.semanticModel.references(srcFile, lineRange.startLine());
-        boolean availableOutSideDeleteRange = false;
-        for (Location location : locations) {
-            if (isWithinLineRange(importDeclarationNode.lineRange(), location.lineRange())) {
-                continue;
+        Optional<Symbol> symbol = semanticModel.symbol(importDeclarationNode);
+
+        if (symbol.isPresent()) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java:116]: We can use early return here to simplify the nesting

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java [hunk 3]
```diff
@@ -190,4 +201,24 @@ public void visit(VariableDeclarationNode variableDeclarationNode) {
             decideVariablesToBeDeleted(variableDeclarationNode.typedBindingPattern().bindingPattern().lineRange());
         }
     }
+
+    @Override
+    public void visit(AnnotationNode annotationNode) {
+        super.visit(annotationNode);
+        Optional<Symbol> annotationNodeSymbol = semanticModel.symbol(annotationNode);
+
+        if (annotationNodeSymbol.isPresent()) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/visitor/UnusedSymbolsVisitor.java:210]: Shall we use early return here and line 213?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java [hunk 4]
```diff
@@ -15,8 +15,26 @@
  */
 package org.ballerinalang.langserver.extensions.ballerina.symbol;
 
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.impl.symbols.BallerinaUnionTypeSymbol;
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java:19]: This implementation class usage has been addressed in a comment above

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java [hunk 5]
```diff
@@ -41,4 +66,79 @@
         return endpoints;
     }
 
+    @Override
+    public CompletableFuture<ExpressionTypeResponse> type(ExpressionTypeRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            ExpressionTypeResponse expressionTypeResponse = new ExpressionTypeResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Optional<Path> filePath = CommonUtil.getPathFromURI(fileUri);
+            if (!filePath.isPresent()) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java:75]: use `filePath.isEmpty()` instead

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java [hunk 6]
```diff
@@ -41,4 +66,79 @@
...
+                return expressionTypeResponse;
+            }
+
+            try {
+                expressionTypeResponse.setDocumentIdentifier(new TextDocumentIdentifier(fileUri));
+
+                // Get the semantic model.
+                Optional<SemanticModel> semanticModel = this.workspaceManager.semanticModel(filePath.get());
+
+                if (semanticModel.isPresent()) {
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java:85]: Using early return will be better if possible.

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

**Review by nadeeshaan — DISMISSED** [review]
LGTM

**Comment by kaviththiranga** [comment]
@NipunaMarcus seems we have PR check failures again

**Review by nadeeshaan — APPROVED** [review]
Low diff coverage and has to be addressed with #30660

**Pull request merged**